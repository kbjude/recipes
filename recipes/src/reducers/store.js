import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '.';

const middlewares = [thunk];

const store = createStore(rootReducer, applyMiddleware(...middlewares));
export default store;
