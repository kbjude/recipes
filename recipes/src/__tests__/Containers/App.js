import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import App from '../../containers/App';

const character = {
  name: 'Breakfast',
  description: 'Breakfast',
};
describe('Display snapshot', () => {
  it('renders correctly a snapshot', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <App
            name={character.strCategory}
            description={character.strCategoryDescription}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
describe('Rendering component', () => {
  it('creates a CharInfo component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <MealDetails
              match={{ params: { id: '52940' } }}
              name={character.strCategory}
              description={character.strCategoryDescription}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Chicken');
  });
});
