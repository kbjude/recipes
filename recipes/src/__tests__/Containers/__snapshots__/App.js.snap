// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display snapshot renders correctly a snapshot 1`] = `
Array [
  <div
    className="sc-hKwCoD hgYXKh"
  >
    <input
      className="sc-eCImvq jMYRuX"
      onChange={[Function]}
      placeholder="SEARCH BY NAME"
      type="text"
    />
  </div>,
  <ul
    className="sc-bdvvaa fawLKJ"
  />,
]
`;
